packages/beacon-state-transition/src/merge/block/processExecutionPayload.ts

```diff
@@ -0,0 +1,107 @@
+import {GAS_LIMIT_DENOMINATOR, MIN_GAS_LIMIT} from "@chainsafe/lodestar-params";
+import {merge, ssz} from "@chainsafe/lodestar-types";
+import {byteArrayEquals, List} from "@chainsafe/ssz";
+import {CachedBeaconState} from "../../allForks";
+import {getCurrentEpoch, getRandaoMix} from "../../util";
+import {ExecutionEngine} from "../executionEngine";
+
+export function isExecutionEnabled(state: merge.BeaconState, body: merge.BeaconBlockBody): boolean {
+  return isMergeBlock(state, body) || isMergeComplete(state);
+}
+
+function isMergeBlock(state: merge.BeaconState, body: merge.BeaconBlockBody): boolean {
+  return (
+    !isMergeComplete(state) &&
+    !ssz.merge.ExecutionPayload.equals(body.executionPayload, ssz.merge.ExecutionPayload.defaultValue())
+  );
+}
+
+function isMergeComplete(state: merge.BeaconState): boolean {
```

Comment:
This is probably just naive implementation, but if we pass in a `TreeBacked<merge.BeaconState>`, we can get a (probably) faster implementation:
```ts
return byteArrayEquals(
  state.latestExecutionPayloadHeader.hashTreeRoot(),
  ssz.merge.ExecutionPayloadHeader.defaultTree.root
);
```

Maybe better saved for future PRs after more benchmarking

Reply:
I'm not sure because the very first property in latestExecutionPayloadHeader is parentHash which will likely be non-zero for any value but the empty one. So in most cases a single byte will be compared. However your solution could trigger much more work if used incorrectly.